### PR TITLE
[BEAM-2299] Run maven install on Windows machine for build/test coverage on Windows.

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
I removed the extra compile-custom goal in runner-api because it was being unused.

### CODE REVIEW

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
Run Seed Job

**Comment by lukecwik** [comment]
Run Java Windows PostCommit

**Comment by lukecwik** [comment]
This starts reproducing the error reported in BEAM-2299:
https://builds.apache.org/job/beam_PostCommit_Java_MavenInstall_Windows/11/testReport/

R: @davorbonaci @jasonkuster

File: sdks/common/runner-api/pom.xml [hunk 1]
```diff
@@ -65,14 +65,11 @@
         <artifactId>protobuf-maven-plugin</artifactId>
         <configuration>
           <protocArtifact>com.google.protobuf:protoc:${protobuf.version}:exe:${os.detected.classifier}</protocArtifact>
-          <pluginId>grpc-java</pluginId>
```
Comment (davorbonaci) [inline sdks/common/runner-api/pom.xml]: Seems unrelated. Separate commit? Justification?

File: .test-infra/jenkins/job_beam_PostCommit_Java_MavenInstall_Windows.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  concurrentBuild()
+
+  // Set common parameters. Note the usage of the Windows label to filter Jenkins executors.
+  common_job_properties.setTopLevelMainJobProperties(delegate, 'master', 100, 'Windows')
+
+  // Set Maven parameters. Note the usage of the Windows Maven installation
+  common_job_properties.setMavenConfig(delegate, 'Maven 3.3.3 (Windows)')
+
+  // Sets that this is a PostCommit job.
+  common_job_properties.setPostCommit(delegate)
```
Comment (davorbonaci) [inline .test-infra/jenkins/job_beam_PostCommit_Java_MavenInstall_Windows.groovy]: This will run it on every push to the repository. Perhaps this one can run less often; say every six 6 hours or something like that.

File: .test-infra/jenkins/common_job_properties.groovy [hunk 3]
```diff
@@ -24,21 +24,33 @@ class common_job_properties {
...
+            label,
+            timeout)
   }
 
   // Sets common top-level job properties. Accessed through one of the above
   // methods to protect jobs from internal details of param defaults.
   private static void setTopLevelJobProperties(context,
                                                String repositoryName,
                                                String defaultBranch,
+                                               String defaultLabel,
```
Comment (davorbonaci) [inline .test-infra/jenkins/common_job_properties.groovy]: defaultLabel -> label (not default, actual)
or, better yet, jenkinsExecutorLabel (ambiguous otherwise)

File: sdks/common/runner-api/pom.xml (hunk 1 above)
Comment (lukecwik) [inline sdks/common/runner-api/pom.xml]: The protoc step runs too quickly and file locking on Windows causes the plugin to fail occasionally complaining of can't copy protoc.exe since the plugin copies protoc.exe to the target folder to launch it.

File: .test-infra/jenkins/common_job_properties.groovy (hunk 3 above)
Comment (lukecwik) [inline .test-infra/jenkins/common_job_properties.groovy]: Done

**Comment by lukecwik** [comment]
Run Seed Job

File: sdks/common/runner-api/pom.xml (hunk 1 above)
Comment (davorbonaci) [inline sdks/common/runner-api/pom.xml]: Ack, but I assume it does a useful operation regardless?

File: .test-infra/jenkins/common_job_properties.groovy (hunk 3 above)
Comment (lukecwik) [inline .test-infra/jenkins/common_job_properties.groovy]: As a side note, I worked on changing the existing names but defaultBranch -> branch conflicts with the usage of 'branch' within the method. Would be nice to use a builder or something instead of a static method that will just grow over time with all the 'variants' that are added. Also default parameters don't work well in Groovy because you can't skip any of them so they become pointless once you need to set an argument.

File: sdks/common/runner-api/pom.xml (hunk 1 above)
Comment (lukecwik) [inline sdks/common/runner-api/pom.xml]: Its a no-op, runner API doesn't define any gRPC services.

**Comment by lukecwik** [comment]
Run Seed Job
